gordon_janitor_gcp/gce_authority.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A GCEAuthority retrieves a list of all instances in all projects that it has
+access to, and which belong to the configured zone. For every project, it will
+create a message containing domain record information and put it into the
+rrset channel. Projects can be filtered by 'project name'. Instances can be
+filtered by tags and metadata.
+
+To use:
+
+.. code-block:: python
+
+    import asyncio
+
+    import gce_authority
+
+    async def run():
+        rrset_channel = asyncio.queue()
+        authority = gce_authority.get_authority(config, rrset_channel)
+        await authority.start()
+        msg = await rrset_channel.get()
+        print(msg)
+
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(run())
+    # prints: {'zone': 'example.com', 'resourceRecords': [...]}
+"""
+
+import asyncio
+import logging
+
+import aiohttp
+import zope.interface
+from gordon_janitor import interfaces
+
+from gordon_janitor_gcp import auth
+from gordon_janitor_gcp import exceptions
+from gordon_janitor_gcp import gcp_clients
+from gordon_janitor_gcp import msg_generators
+
+
+class GCEAuthorityBuilder:
```

Comment:
Hmm, I'm starting to dislike a lot of this divergence of style... I'd like to talk about this Monday AM.

Reply:
Sure! Just lemme know whenever's good.

Reply:
Talked about this in person :tada: